### PR TITLE
ZMQ: add a command line option to disable the ZMQ server

### PR DESCRIPTION
**Pull request by jtgrassie**

### CODE REVIEW

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -170,25 +171,30 @@ bool t_daemon::run(bool interactive)
...
 
-      for(auto& rpc : mp_internals->rpcs)
-        rpc->stop();
+        for(auto& rpc : mp_internals->rpcs)
+          rpc->stop();
 
-      return false;
-    }
+        return false;
+      }
```
Comment (moneromooo-monero) [inline src/daemon/daemon.cpp]: That looks like a very diff-spammy way to say: if (!zmq_rpc_disabled && !zmq_server.addTcpSocket(...))

Comment (jtgrassie) [reply src/daemon/daemon.cpp]: Not really, and I nearly did what you are suggesting. The addTcpSocket failure (if check) is independent to deciding to disable zmq.

Comment (jtgrassie) [reply src/daemon/daemon.cpp]: I can rejig it if you are really bugged by it, but then of course I also need to wrap zmq_server.run(); in an if block.

Comment (moneromooo-monero) [reply src/daemon/daemon.cpp]: OK, there's a message that's extra. Fine :)

**Pull request merged**